### PR TITLE
api: Fix compatibility with pre 0.9 API jobs

### PR DESCRIPTION
**Pull request by dantoml**
Prior to 0.9b, it was possible to submit API jobs with `nil` blocks. For
example in a docker job, you could specify something like:

```json
{
        "config": {
                "image": "bash:3",
                "mounts": null,
        }
}
```

which would be evaluated as equivalent to something like:

```hcl
config {
        image = "bash:3"
        mounts /* missing body */
}
```

In 0.9b1-2, this would cause a validation error when encoding the
configuration for the driver.

This PR reintroduces support for explicitly nilling block fields in the
API, by updating HCL2 json mapping to support this. https://github.com/hashicorp/hcl2/pull/87

We also add a regression test that explicitly bounces a nil `mounts`
specification through `ParseHclInterface`.

### CODE REVIEW

File: drivers/docker/config_test.go [hunk 1]
```diff
@@ -41,6 +42,32 @@ func TestConfig_ParseHCL(t *testing.T) {
 	}
 }
 
+func TestConfig_ParseJSON_Regression(t *testing.T) {
+	cases := []struct {
+		name  string
+		input map[string]interface{}
```
Comment (notnoop) [inline drivers/docker/config_test.go]: I think the test would be much better if we specify what the expected resulting values are, not just that it parses without errors.

File: drivers/docker/config_test.go [hunk 2]
```diff
@@ -41,6 +42,32 @@ func TestConfig_ParseHCL(t *testing.T) {
...
+	require.Empty(t, diag)
+
+	for _, c := range cases {
+		c := c
+		t.Run(c.name, func(t *testing.T) {
+			// TODO(dani): Expand these tests to validate the produced TaskConfig
+			//             when bouncing through messagepack.
+			v, diag := hclutils.ParseHclInterface(c.input, spec, nil)
+			require.Empty(t, diag)
+			require.NotNil(t, v)
```
Comment (notnoop) [inline drivers/docker/config_test.go]: You can use the helper function for msgpack stuff like, like how `TestConfig_ParseHCL` test does but it uses json text input though:

```
cfgStr := `{"config": {"image": "bash:4", "mounts": nil}}`
var tc *TaskConfig
hclutils.NewConfigParser(taskConfigSpec).ParseJson(t, cfgStr, &tc)
```

Comment (dantoml) [reply drivers/docker/config_test.go]: Sweet I had trouble with manually handling messagepack for some non obvious reasons - updated to use the helper

File: drivers/docker/config_test.go [hunk 3]
```diff
@@ -41,6 +41,34 @@ func TestConfig_ParseHCL(t *testing.T) {
 	}
 }
 
+func TestConfig_ParseJSON_Regression(t *testing.T) {
```
Comment (notnoop) [inline drivers/docker/config_test.go]:
```suggestion
func TestConfig_ParseJSON(t *testing.T) {
```

**Pull request merged**